hack/install-vm.sh

```diff
@@ -0,0 +1,407 @@
+    if [[ -z "$AGENT_NAMESPACE" ]]; then
+        AGENT_NAMESPACE="default"
+    fi
+    echo "Updating externalNodeNamespace to $AGENT_NAMESPACE"
+    sed -i "s|#externalNodeNamespace: default|externalNodeNamespace: $AGENT_NAMESPACE|g" $AGENT_CONF_PATH/antrea-agent.conf
+    echo "Updating ovsBridge to $OVS_BRIDGE"
+    sed -i "s|#ovsBridge: br-int|ovsBridge: $OVS_BRIDGE|g" $AGENT_CONF_PATH/antrea-agent.conf
+}
+​
+start_antrea_agent_service() {
+    if [[ ! -f "$AGENT_BIN_PATH" ]]; then
+        echoerr "Error $AGENT_BIN_PATH file not found"
+        exit 1
+    fi
+    mkdir -p $AGENT_LOG_DIR
+    mkdir -p $INSTALL_PATH
+    cp "$AGENT_BIN_PATH" "$INSTALL_PATH"
+    echo "Copying $BASH_SOURCE to ${AGENT_CONF_PATH}/install-vm.sh"
+    cp "$BASH_SOURCE" "${AGENT_CONF_PATH}/install-vm.sh"
+    chmod +x "${AGENT_CONF_PATH}/install-vm.sh"
+    cat >/etc/systemd/system/antrea-agent.service << EOF
+[Unit]
+Description="antrea-agent as a systemd service"
+After=network.target
+[Service]
+Environment="NODE_NAME=$NODE_NAME"
+ExecStartPre=${AGENT_CONF_PATH}/install-vm.sh --validate-ovs --ovs-bridge $OVS_BRIDGE
+ExecStart=$INSTALL_PATH/antrea-agent \
+--config=$AGENT_CONF_PATH/antrea-agent.conf \
+--logtostderr=false \
+--log_file=$AGENT_LOG_DIR/antrea-agent.log
+Restart=on-failure
+[Install]
+WantedBy=multi-user.target
+EOF
+    systemctl daemon-reload
+    systemctl enable "$ANTREA_AGENT"
+    echo "Starting ${ANTREA_AGENT} service"
+    systemctl start "$ANTREA_AGENT"
+    systemctl status "$ANTREA_AGENT"
+}
+​​
+# Generic function to execute command with timeout
+execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+	      echoerr "$cmd failed with status $ret at $(date)"
+        echoerr "Exiting!! please check internet connectivity and re-run install script"
+        exit 2
+    fi
+}
+​
+# Install docker as container runtime
+install_container_runtime() {
+    os=`grep -Po '^ID=\K.*' /etc/os-release | sed -e 's/^"//' -e 's/"$//'`
+    echo "Installing container runtime on $os"
+    case "$os" in
+        ubuntu)
+	    # Uninstall older versions of Docker(called docker, docker.io, or docker-engine)
+            timeout --preserve-status --foreground 1m apt-get remove docker docker-engine docker.io containerd runc
+            timeout --preserve-status --foreground 1m apt-get update
+	    cmd="apt-get update"
+	    execute_command "$cmd" 1m
+	    cmd="apt-get install -y ca-certificates curl gnupg lsb-release"
+	    execute_command "$cmd" 1m
+	    # Add docker's offical GPG key
+	    rm -rf /usr/share/keyrings/docker-archive-keyring.gpg
+	    timeout --preserve-status --foreground 10s curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
```

Comment:
Change https://download.docker.com/linux/ubuntu/gpg to variable.

Changee /usr/share/keyrings/docker-archive-keyring.gpg to variable.

Also since the script is running as a sudo user, so you can remove the sudo from here.